tests/test_series.py

```diff
@@ -732,6 +732,19 @@ def test_cat_accessor() -> None:
     # GH 43
     s = pd.Series(pd.Categorical(["a", "b", "a"], categories=["a", "b"]))
     check(assert_type(s.cat.codes, "pd.Series[int]"), pd.Series, int)
+    # GH 139
+    ser = pd.Series([1, 2, 3], name="A").astype("category")
+    check(assert_type(ser.cat.set_categories([1, 2, 3]), pd.Series), pd.Series)
```

Comment:
Not familiar with how generic Series is. Do you still have access to the type of the elements here?

Reply:
> Not familiar with how generic Series is. Do you still have access to the type of the elements here?

Yes, it will be the underlying type of the categories.  We can't do `Series[category]`, because categories could be strings, ints, etc.

Nevertheless, I changed things to test the type of the value inside the `Series` in the new commit.

Reply:
Okay! In general, I would be fine with only adding runtime checks to ensure the type annotations. In this case, I wouldn't mind removing  the checks for `int` as `int` isn't part of the type annotation.